app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/PebbleSync.java

```diff
@@ -36,16 +40,48 @@
     private Context mContext;
     private BgGraphBuilder bgGraphBuilder;
     private BgReading mBgReading;
+    private static int lastTransactionId;
+    public PebbleSync(Context context){
+        this.mContext = context;
+        mBgReading = null;
+        init();
+    }
+    private void init() {
+        Log.i(TAG, "Initialising...");
+        Log.i(TAG, "configuring PebbleDataReceiver");
+
+        PebbleKit.registerReceivedDataHandler(mContext, new PebbleKit.PebbleDataReceiver(PEBBLEAPP_UUID) {
+            @Override
+            public void receiveData(final Context context, final int transactionId, final PebbleDictionary data) {
+                Log.d(TAG,"receiveData: transactionId is "+String.valueOf(transactionId));
+                if(lastTransactionId == 0 || transactionId != lastTransactionId) {
+                    lastTransactionId = transactionId;
+                    Log.d(TAG, "Received Query. data: " + data.size());
+                    PebbleKit.sendAckToPebble(context, transactionId);
+                    sendData(context, mBgReading);
```

Comment:
My concern is that in the BgSendQueue we call the sendData function, but then we call it again here, and Im not sure if that is expected, but like I said, I dont know a whole lot about what the receiveData handler is